src/Sqs/SqsJob.php

```diff
@@ -44,8 +44,24 @@ public function payload()
 
         $payload['job'] = $this->handler;
 
-        $payload['data'] = unserialize($payload['Message']);
+        if ($this->isJson($payload['Message']) === true) {
+            $payload['data'] = json_decode($payload['Message'], true);
+        } else {
+            $payload['data'] = unserialize($payload['Message']);
+        }
 
         return $payload;
     }
+
+    /**
+     * @param $message
+     *
+     * @return bool
+     */
+    private function isJson($message)
+    {
+        $result = json_decode(trim($message, '"'), true);
+
+        return is_array($result);
+    }
```

Comment:
Nice catch. I think it would be good if we removed the typing from it in the first place, because like you said. there might be multiple workers/subscribers that are listening to the same thing, it will all make sense to use protocol buffer in the both ends.

Happy to take the `serialize` check in Both `Subscriber` and `SNS` publisher too.

Reply:
how long will it take? it stopped my task) thank you so much)

Reply:
Hello)) can you approve pull request?) Or should I do something?)

Reply:
Hey @kak2z7702, I will approve and mrege